replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/StatusCondenser.java

```diff
@@ -0,0 +1,183 @@
+    this.scheduledExecutor = notNull(scheduledExecutor);
+    this.security = security;
+  }
+
+  public void init() {
+    scheduledExecutor.scheduleAtFixedRate(
+        this::condenseResultsAsSystemUser, 1, 60, TimeUnit.MINUTES);
+  }
+
+  public void destroy() {
+    scheduledExecutor.shutdownNow();
+  }
+
+  void condenseResultsAsSystemUser() {
+    security.runAsAdmin(
+        () -> {
+          try {
+            security.runWithSubjectOrElevate(
+                () -> {
+                  replicatorConfigLoader.getAllConfigs().stream().forEach(this::condenseResults);
+                  return null;
+                });
+          } catch (SecurityServiceException | InvocationTargetException e) {
+            LOGGER.error("Error condensing replication history.", e);
+          }
+          return null;
+        });
+  }
+
+  void condenseResults(ReplicatorConfig config) {
+    List<ReplicationStatus> events =
+        Lists.reverse(
+            replicatorHistory
+                .getReplicationEvents(config.getName())
+                .stream()
+                .filter(s -> s.getPeriod().equals(StatusPeriod.INSTANCE))
+                .collect(Collectors.toList()));
+    condenseResults(events, StatusPeriod.DAY, 1);
+    events =
+        Lists.reverse(
+            replicatorHistory
+                .getReplicationEvents(config.getName())
+                .stream()
+                .filter(s -> s.getPeriod().equals(StatusPeriod.DAY))
+                .collect(Collectors.toList()));
+    condenseResults(events, StatusPeriod.WEEK, 7);
+    events =
+        Lists.reverse(
+            replicatorHistory
+                .getReplicationEvents(config.getName())
+                .stream()
+                .filter(s -> s.getPeriod().equals(StatusPeriod.WEEK))
+                .collect(Collectors.toList()));
+    condenseResults(events, StatusPeriod.YEAR, 365);
+  }
+
+  /**
+   * Condenses events into new events of the given type based on the given window
+   *
+   * @param events events to condense. Events must be ordered oldest to newest
+   * @param outType the output event type
+   * @param daysInWindow number of days in the outputType
+   */
+  void condenseResults(List<ReplicationStatus> events, StatusPeriod outType, int daysInWindow) {
+    ReplicationStatus oldest = null;
+    Date cutoff = null;
+    Set<String> idsToRemove = new HashSet<>();
+    for (ReplicationStatus status : events) {
+      if (oldest == null) {
+        oldest = new ReplicationStatus(status.getReplicatorName());
+        oldest.setPeriod(outType);
+        oldest.setStartTime(startOfDay(status.getStartTime()));
+        cutoff = new Date(oldest.getStartTime().getTime() + TimeUnit.DAYS.toMillis(daysInWindow));
+      }
+
+      if (status.getStartTime().before(cutoff)) {
+        addStats(oldest, status);
+        idsToRemove.add(status.getId());
+      } else {
+        // save oldest day as a week
+        replicatorHistory.addReplicationEvent(oldest);
```

Comment:
❓ Couldn't this also be the oldest week saved as a year? Above you seem to pass in different time units.